src/blas/KokkosBlas1_iamax.hpp

```diff
@@ -0,0 +1,159 @@
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOSBLAS1_IAMAX_HPP_
+#define KOKKOSBLAS1_IAMAX_HPP_
+
+#include<KokkosBlas1_iamax_spec.hpp>
+#include<KokkosKernels_helpers.hpp>
+
+namespace KokkosBlas {
+
+/// \brief Return the (smallest) index of the element of the maximum magnitude of the vector x. 
+///
+/// \tparam XVector Type of the first vector x; a 1-D Kokkos::View.
+///
+/// \param x [in] Input 1-D View.
+///
+/// \return The (smallest) index of the element of the maximum magnitude; a single value.
+template<class XVector>
+typename XVector::size_type iamax (const XVector& x)
+{
+  static_assert (Kokkos::Impl::is_view<XVector>::value,
+                 "KokkosBlas::iamax: XVector must be a Kokkos::View.");
+  static_assert (XVector::rank == 1, "KokkosBlas::iamax: "
+                 "Both Vector inputs must have rank 1.");
+
+  typedef typename XVector::size_type index_type;
+
+  typedef Kokkos::View<typename XVector::const_value_type*,
+    typename KokkosKernels::Impl::GetUnifiedLayout<XVector>::array_layout,
+    typename XVector::device_type,
+    Kokkos::MemoryTraits<Kokkos::Unmanaged> > XVector_Internal;
+
+  typedef Kokkos::View<index_type,
+    Kokkos::LayoutLeft,
+    Kokkos::HostSpace,
+    Kokkos::MemoryTraits<Kokkos::Unmanaged> > RVector_Internal;
+
+  index_type result;
+  RVector_Internal R = RVector_Internal(&result);
+  XVector_Internal X = x;
+
+  Impl::Iamax<RVector_Internal,XVector_Internal>::iamax (R,X);
+  Kokkos::fence();
+  return result;
+}
+
+/// \brief R(j) = iamax(X(i,j))
+///
+/// Replace each entry in R with the (smallest) index of the element of the maximum magnitude of the
+/// corresponding entry in X.
+///
+/// \tparam RMV 1-D or 2-D Kokkos::View specialization.
+/// \tparam XMV 1-D or 2-D Kokkos::View specialization.  It must have
```

Comment:
The documentation here doesn't match the "brief" description.  R can't possibly have the same rank as X.

Reply:
Thanks @mhoemmen . It is fixed.